iot-hub/Samples/device/DeviceReconnectionSample/RetryOperationHelper.cs

```diff
@@ -27,7 +27,7 @@ internal class RetryOperationHelper
         /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
         internal static async Task RetryTransientExceptionsAsync(
             string operationName,
-            Func<Task> asyncOperation,
+            Func<Task<T>> asyncOperation,

```

Comment:
Why was this change to `Task<T>`? On line 39 we call it, but we don't check the result, so I wonder why `Task` alone as a return type is insufficient.

Reply:
I planned to call `s_deviceClient.GetTwinAsync()` as `asyncOperation` in **DeviceReconnectionSample** class, so made it with a generic return type here. But I think you are right as I don't necessarily need a value to be returned, instead I just need this value itself.